### ISSUE TITLE
intregrate ci with dreck

### ISSUE THREAD

**Issue by miekg** [coredns/ci#63]
The release script is working (almost 100%, next release will pinpoint remaining failure). Good thing is except the caddy's config all just script come from github.

We should move ci to this as well, which means we can remove:
* remove a bunch of shell scripting, as dreck can handle some of this, i.e. updating the PR status, etc. (not 100% sure about the details link though)
* remove webhook and it's config

**Comment by miekg** [comment]
this will become critical soon, because I'm going to clean up all the webhook stuff. Might also require (small) changes to miekg/dreck to be fully functional.

**Comment by miekg** [comment]
This is the current caddy config that calls webhook:
~~~ caddyfie
    proxy /hooks/coredns/benchmark/pull     localhost:9000/hooks/coredns-benchmark-pull     { without /hooks/coredns/benchmark/pull }
    proxy /hooks/coredns/comment            localhost:9000/hooks/coredns-comment            { without /hooks/coredns/comment }
~~~
this is the webhook.conf (minus passwords)
~~~ txt
[
  {
    "id": "coredns-benchmark-pull",
    "execute-command": "/opt/bin/coredns-benchmark-pull",
    "command-working-directory": "/var/www",
    "pass-environment-to-command":
    [
      {
        "source": "entire-payload",
        "envname": "PAYLOAD"
      }
    ],
    "trigger-rule":
    {
      "match":
        {
          "type": "payload-hash-sha1",
          "secret": "XXXXXX",
          "parameter":
        {
          "source": "header",
          "name": "X-Hub-Signature"
        }
      }
    }
  },

  {
    "id": "coredns-comment",
    "execute-command": "/opt/bin/pr-comment-hook",
    "command-working-directory": "/var/www",
    "pass-environment-to-command":
    [
      {
        "source": "entire-payload",
        "envname": "PAYLOAD"
      }
    ],
    "trigger-rule":
    {
      "match":
        {
          "type": "payload-hash-sha1",
          "secret": "XXXXXX",
          "parameter":
        {
          "source": "header",
          "name": "X-Hub-Signature"
        }
      }
    }
  },

  {
    "id": "coredns-comment-test",
    "execute-command": "/opt/bin/pr-comment-hook-fork.sandeep",
    "command-working-directory": "/var/www",
    "pass-environment-to-command":
    [
      {
        "source": "entire-payload",
        "envname": "PAYLOAD"
      }
    ],
    "trigger-rule":
    {
      "match":
        {
          "type": "payload-hash-sha1",
          "secret": "XXX",
          "parameter":
        {
          "source": "header",
          "name": "X-Hub-Signature"
        }
      }
    }
  }

]
~~~

**Comment by miekg** [comment]
I will remove this setup somewhere in the next week from drone and only leave the release infra on that machine. I.e. this stuff has to move to a different machine.

Had dreck integration be attempted or done by now I would have moved this myself, but now I'm just going to remove it. This means the kube ci will probably not work.

**Comment by chrisohaver** [comment]
@miekg, How can I help to make this go more smoothly without loss of the existing coredns CI coverage?

@fturib @johnbelamaric

**Comment by chrisohaver** [comment]
@miekg, can you point me in the right direction for getting this integrated so we don't lose the existing coredns/ci coverage? What do I need to do?

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/ci] intregrate ci with..." ]
>@miekg, How can I help to make this go more smoothly without loss of the existing coredns CI coverage?
>
>@fturib @johnbelamaric

I *think* things should just work with *dreck*, it should use the new github api 
to provide callbacks to the API (i.e. what we have now). I.e. you can the script 
from *dreck*.

I don't now what to do with the logs link we have right now; out-of-the-box we 
don't have a way to display those.

Alternatively you can setup a new host and port the stuff over - as is - and we 
point to github to that host for the ci test.

Longer term, dreck integration would useful.

Let me at least double check how you can call a script from github, via dreck, 
because that should work, but I can't remember how I set it up.

**Comment by chrisohaver** [comment]
@miekg, started looking into this today... do you have miekg/dreck installed on drone?  or is it somewhere else? It appears that my ssh keys are no longer on drone.coredns.io.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/ci] intregrate ci with..." ]
>@miekg, started looking into this today... do you have miekg/dreck installed on drone?  or is it somewhere else? It appears that my ssh keys are no longer on drone.coredns.io.

I've removed all external access to this machine

**Comment by chrisohaver** [comment]
> I've removed all external access to this machine

OK, is this where dreck lives?  How to we update it?

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/ci] intregrate ci with..." ]
>> I've removed all external access to this machine
>
>OK, is this where dreck lives?  How to we update it?

What I can say: I've seen weird access to this host, related to kubernetes 
stuff. I've blocked all access from infoblox folks

**Comment by chrisohaver** [comment]
> I've blocked all access from infoblox folks

okaaaay ... thanks for letting us know? ... I'll set up something on another system then...

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/ci] intregrate ci with..." ]
>> I've blocked all access from infoblox folks
>
>okaaaay ... thanks for letting us know? ... I'll set up something on another system then...

I've email security and chris@coredns.io, there things stop.

And insecurely accessing things is just a big no, no. So yes, then infoblox is 
left to deal with it, because such a breach of trust is unforgivable.

**Comment by chrisohaver** [comment]
FWIW, I'm not on the security mailing list... and I dont know how to access chris@coredns.io mail... i thought it was forwarded to cohaver@infoblox.com, but apparently not...

I want to understand what I did wrong so i don't do it again... I've been using ssh to access the system. I have had to occasionally go in to bounce docker (it gets wedged occasionally, breaking ci), and to update the ci webhook script.  We also made some changes recently to try to handle ci tests from forks (to you don't have to see "Test PR" noise) - although, this became complicated and we had to set it aside.
